stdlib/public/runtime/Errors.cpp

```diff
@@ -78,17 +80,27 @@ static bool getSymbolNameAddr(llvm::StringRef libraryName, SymbolInfo syminfo,
   // demangle with swift. We are taking advantage of __cxa_demangle actually
   // providing failure status instead of just returning the original string like
   // swift demangle.
-  int status;
-  char *demangled = abi::__cxa_demangle(syminfo.symbolName, 0, 0, &status);
-  if (status == 0) {
-    assert(demangled != nullptr && "If __cxa_demangle succeeds, demangled "
-                                   "should never be nullptr");
-    symbolName += demangled;
-    free(demangled);
-    return true;
-  }
-  assert(demangled == nullptr && "If __cxa_demangle fails, demangled should "
-                                 "be a nullptr");
+  
+#if defined(_WIN32)
+    char demangled[1024];
+    DWORD length = UnDecorateSymbolName(syminfo.symbolName, demangled, 1024, 0);
```

Comment:
Using `sizeof(demanglled) / sizeof(*demangled)` or defining a local `array_size` as follows would be nicer:

```
template <typename T, size_t N>
size_t array_size(const T (&)[N]) { return N; }
```